zipline/gens/brokers/alpaca_broker.py

```diff
@@ -0,0 +1,266 @@
+        results = {}
+        for abars in bars_list:
+            results[abars.symbol] = abars.bars
+        return results
+
+    @property
+    def positions(self):
+        z_positions = zp.Positions()
+        positions = self._api.list_positions()
+        position_map = {}
+        symbols = []
+        for pos in positions:
+            symbol = pos.symbol
+            try:
+                z_position = zp.Position(symbol_lookup(symbol))
+            except SymbolNotFound:
+                continue
+            z_position.amount = pos.qty
+            z_position.cost_basis = float(pos.cost_basis)
+            z_position.last_sale_price = None
+            z_position.last_sale_date = None
+            z_positions[symbol_lookup(symbol)] = z_position
+            symbols.append(symbol)
+            position_map[symbol] = z_position
+
+        quotes = self._api.list_quotes(symbols)
+        for quote in quotes:
+            price = quote.last
+            dt = quote.last_timestamp
+            z_position = position_map[quote.symbol]
+            z_position.last_sale_price = float(price)
+            z_position.last_sale_date = dt
+        return z_positions
+
+    @property
+    def portfolio(self):
+        account = self._api.get_account()
+        z_portfolio = zp.Portfolio()
+        z_portfolio.cash = account.cash
+        z_portfolio.positions = self.positions
+        z_portfolio.positions_value = float(
+            account.portfolio_value) - float(account.cash)
+        return z_portfolio
+
+    @property
+    def account(self):
+        account = self._api.get_account()
+        z_account = zp.Account()
+        z_account.buying_power = account.cash
+        z_account.total_position_value = float(
+            account.portfolio_value) - float(account.cash)
+        return z_account
+
+    @property
+    def time_skew(self):
+        return pd.Timedelta('0 sec')  # TODO: use clock API
+
+    def _new_order_id(self):
+        return uuid.uuid4().hex
+
+    def order(self, asset, amount, style):
+        symbol = asset.symbol
+        qty = amount
+        side = 'buy' if amount > 0 else 'sell'
+        order_type = 'market'
+        if isinstance(style, MarketOrder):
+            order_type = 'market'
+        elif isinstance(style, LimitOrder):
+            order_type = 'limit'
+        elif isinstance(style, StopOrder):
+            order_type = 'stop'
+        elif isinstance(style, StopLimitOrder):
+            order_type = 'stop_limit'
+
+        limit_price = style.get_limit_price(side == 'buy') or None
+        stop_price = style.get_stop_price(side == 'buy') or None
+
+        zp_order_id = self._new_order_id()
+        dt = pd.to_datetime('now', utc=True)
+        zp_order = ZPOrder(
+            dt=dt,
+            asset=asset,
+            amount=amount,
+            stop=stop_price,
+            limit=limit_price,
+            id=zp_order_id,
+        )
+
+        order = self._api.submit_order(
```

Comment:
Question: Robinhood needed all prices rounded to the next 5cts, not the case with Alpaca?

Reply:
For submitting order?  No, we don't have such rule.